Gruntfile.js

```diff
@@ -0,0 +1,352 @@
+module.exports = function(grunt) {
+
+    var fs = require('fs');
+
+    var awsConfig = {
+        accessKey: false,
+        secretKey: false,
+        region: false,
+        s3Bucket: false
+    };
+
+    // Breaks out the keys and also gives us a nice reference of what
+    // needs to be in here without being able to be changed.
+    var awsConfigKeys = Object.keys(awsConfig);
+
+    // Take inputs for the awsConfig, prefixed with 'aws.'.
+    // For example accessKey can be inputted with '--aws.accessKey=accessKey'
+    awsConfigKeys.forEach(function(key) {
+        var optionKey = 'aws.' + key;
+        var optionValue = grunt.option(optionKey);
+        if (optionValue) {
+            // Use the original key to update it.
+            awsConfig[key] = optionValue;
+        }
+    });
+
+    grunt.file.move = function(from,to) {
+        grunt.log.writeln('Moved '+from.cyan+' to '+to.cyan);
+        grunt.file.copy(from,to);
+        grunt.file.delete(from);
+    };
+
+    grunt.loadNpmTasks('grunt-contrib-compress');
+    grunt.loadNpmTasks('grunt-contrib-requirejs');
+    grunt.loadNpmTasks('grunt-contrib-watch');
+    grunt.loadNpmTasks('grunt-version');
+    grunt.loadNpmTasks('grunt-aws');
+    grunt.loadNpmTasks('grunt-fastly');
+    grunt.loadNpmTasks('grunt-git');
+
+    grunt.initConfig({
+        aws: awsConfig,
+        version: {
+            options: {
+                pkg: grunt.file.readJSON('package.json')
+            },
+            header: {
+                options: {
+                    prefix: '@version *'
+                },
+                src: ['dist/plugins/**/*.js']
+            },
+            constant: {
+                options: {
+                    prefix: 'VERSION\\s*:\\s*[\'"]'
+                },
+                src: ['dist/plugins/**/*.js']
+            }
+        },
+        watch: {
+            js : {
+                files: ['./src/*.js'],
+                tasks: ['compileJS'],
+                options: {
+                    spawn: false,
+                    debounceDelay: 250,
+                    interrupt: true
+                }
+            }
+        },
+        requirejs: {
+            compile_eloqua_plugin: {
+                options: {
+                    baseUrl: './src',
+                    include: ['eloqua'],
+                    optimize: 'none',
```

Comment:
Yeah so I went back and forth on this. On the one hand it it's a slightly smaller file if we optimize it. On the other hand, we do publish a gzip'd version, so the memory savings from optimizing would probably be minimal. My bigger concern was that since this is a public repo, it feels kind of sketchy to distribute an optimized version through the CDN - consumers no longer have a way to verify that what's in this repo is what they're actually pulling down from sdk.ceros.com.

I chatted with Abent about it and he also felt that if we gzip, there's not any real need for optimization. If you disagree I'm open to changing my mind!